src/main/java/com/dft/prestashop/PrestashopSDK.java

```diff
@@ -0,0 +1,98 @@
+package com.dft.prestashop;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import static com.dft.prestashop.constantcodes.ConstantCode.AUTHORIZATION;
+import static com.dft.prestashop.constantcodes.ConstantCode.BASE_ENDPOINT;
+import static com.dft.prestashop.constantcodes.ConstantCode.BASIC;
+import static com.dft.prestashop.constantcodes.ConstantCode.HTTP;
+
+public class PrestashopSDK {
+
+    private final HttpClient client;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private final PrestashopAccessCredentials prestashopAccessCredentials;
+    int MAX_ATTEMPTS = 50;
+    int TIME_OUT_DURATION = 60000;
+
+    public PrestashopSDK(String domain, String webServiceKey) {
+        prestashopAccessCredentials = new PrestashopAccessCredentials(domain, encodeToBase64(webServiceKey));
+        client = HttpClient.newHttpClient();
+    }
+
+
+    @SneakyThrows
+    protected URI baseUrl(String path) {
+        return new URI(new StringBuilder()
```

Comment:
remove StringBuilder and use String.formate

Reply:
removed StringBuilder and used String.formate() function in baseUrl() function